tools/generators/legacy/src/DTO/XcodeScheme.swift

```diff
@@ -61,62 +68,79 @@ disabled, but the target is referenced in the scheme's \(keyPath.actionType) act
+        } else if let launchAction = launchAction,
+            buildTargets[launchAction.target]?.buildFor.profiling != .disabled
+        {
+            newProfileAction = .init(
+                target: launchAction.target,
+                buildConfigurationName: launchAction.buildConfigurationName,
+                workingDirectory: launchAction.workingDirectory
+            )
+        } else {
+            newProfileAction = nil
+        }
 
-            // Enable archiving for any targets that have running enabled
-            try buildTargets.values.filter(\.buildFor.running.isEnabled).map(\.label).forEach {
-                try enableBuildForValue($0, \.archiving, .setIfAble)
-            }
+        // Update the buildFor for the build targets
+        try testAction?.targets.forEach { try enableBuildForValue($0, \.testing) }
+        try launchAction.map { try enableBuildForValue($0.target, \.running) }
+        try newProfileAction.map { try enableBuildForValue($0.target, \.profiling) }
 
-            // Enable analyze for all targets
-            try buildTargets.keys.forEach { try enableBuildForValue($0, \.analyzing, .setIfAble) }
+        // If no build targets have running enabled, then enable it for all targets
+        if !buildTargets.values.contains(where: \.buildFor.running.isEnabled) {
+            try buildTargets.keys.forEach { try enableBuildForValue($0, \.running) }
+        }
 
-            // Create a new build action which includes all of the referenced labels as build targets
-            // We must do this after processing all of the other actions.
-            let newBuildAction = try XcodeScheme.BuildAction(
-                targets: buildTargets.values,
-                preActions: buildAction?.preActions ?? [],
-                postActions: buildAction?.postActions ?? []
-            )
+        // Enable archiving for any targets that have running enabled
+        try buildTargets.values.filter(\.buildFor.running.isEnabled).map(\.label).forEach {
+            try enableBuildForValue($0, \.archiving, .setIfAble)
+        }
 
-            return try .init(
-                name: name,
-                buildAction: newBuildAction,
-                testAction: testAction,
-                launchAction: launchAction,
-                profileAction: newProfileAction
+        // Enable analyze for all targets
+        try buildTargets.keys.forEach { try enableBuildForValue($0, \.analyzing, .setIfAble) }
+
+        // Create a new build action which includes all of the referenced labels as build targets
+        // We must do this after processing all of the other actions.
+        let newBuildAction = try XcodeScheme.BuildAction(
+            targets: buildTargets.values,
+            preActions: buildAction?.preActions ?? [],
+            postActions: buildAction?.postActions ?? []
+        )
+
+        // Default TestAction
+        let newTestAction: XcodeScheme.TestAction?
```

Comment:
This is effectively the new thing in this scope, `testAction?.targets` needs to be expanded if `aliasLabels` is not empty

Compared `buildTargets.values.map { $0.label }` with `testAction?.targets + expansions from aliasLabels` and it's the same list so I assume it's safe to override `targets` with values from `buildTargets` directly here since that's where the expansion happened above in `enableBuildForValue`.

Reply:
You'll need to move the `try testAction?.targets.forEach { try enableBuildForValue($0, \.testing) }` line to be after this, so it does it based on `newTestAction`.

Reply:
Ok, so will be explicit here and use `testAction?.targets` but expanding what is in `aliasLabels` only